python/mxnet/ndarray/numpy/linalg.py

```diff
@@ -352,3 +352,57 @@ def slogdet(a):
     (1., -1151.2925464970228)
     """
     return _npi.slogdet(a)
+
+
+def solve(a, b):
+    r"""
+    Solve a linear matrix equation, or system of linear scalar equations.
+
+    Computes the "exact" solution, `x`, of the well-determined, i.e., full
+    rank, linear matrix equation `ax = b`.
+
+    Parameters
+    ----------
+    a : (..., M, M) ndarray
+        Coefficient matrix.
+    b : {(..., M,), (..., M, K)}, ndarray
+        Ordinate or "dependent variable" values.
+
+    Returns
+    -------
+    x : {(..., M,), (..., M, K)} ndarray
+        Solution to the system a x = b.  Returned shape is identical to `b`.
+
+    Raises
+    ------
+    LinAlgError
```

Comment:
Are we really raising this same error when input is singular or not square?

Reply:
If the input is singular, cblas function will raise error. If the input is not square, shape inference will raise error

Reply:
I mean are we raising an error of type `MXNetError` or `LinAlgError`?

Reply:
MXNetError

Reply:
So you need to change the doc to reflect that

Reply:
Ok